Js2

JoranEngelund opened the pull request:
_It has been developed by [Benjamin Løndal](https://github.com/dvergnir) & [Jøran Engelund](https://github.com/JoranEngelund) as part of the JavaScript 2 Course Assignment at Noroff._

**Implemented features:** 

- Login system
- Register profile system
- Create/Read/React/Comment/Update/Delete posts
- Search on posts
- Edit profile media
- Follow/Unfollow other profiles
- Sort/Filtering of posts
- View specific post by ID

JoranEngelund committed:
Created registration page with form

Added: form with inputs, regex, title for better interaction design
Added: navigation link to login page, for better user-flow

JoranEngelund committed:
Updated inputs with regex pattern in line with API documentation

JoranEngelund committed:
updated create post form ->

Added: title input (required)
Added: tags input (optional)
Added: media input (optional)

JoranEngelund committed:
Updated and implemented new UI on home.html ->

Added: Title input(Required), Tags input(optional) media input(optional)  for create post form.
Added: UI component for updating post. with modal for updating the inputs in the post
Added: Delete component for posts, so user can delete their posts.

JoranEngelund committed:
Updated/implemented new UI for profile ->

Added: Title input(required), tags input(optional), media-url input(optional) for create post form on profile wall

Added: UI component on post to update posts, aswell as modal that contains the form with the inputs you can update.
Added: Delete component, so user can delete their posts.

JoranEngelund committed:
separated styles for card-img ->

Change: Separated styling for card-img-top/bottom.
Added: max-width 40% on card-img-bottom

JoranEngelund committed:
added static image, for post media

JoranEngelund committed:
Implemented new UI and features for home.html ->

Added: link for user name on post
Added: card-title for posts
Added: link for viewing single post
Added: Edit link with corresponding modal for editing post details
Added: Delete links for posts

JoranEngelund committed:
Created post-specific page ->

Created: post-specific page for viewing specific posts by ID.
Added: Modal for editing posts aswell

JoranEngelund committed:
Added new UI features for profile ->

Added: cta for editing profile media, aswell as a corresponding modal with input to change the media
Added: CTA for follow profile
Added: Edit link for updating post details
Added: Delete link, for deleting post

JoranEngelund committed:
added class and styling for time stamp on posts

JoranEngelund committed:
Added UI for time-stamps on posts across all pages

JoranEngelund committed:
removed searchbar from header

JoranEngelund committed:
removed searchbar from header

JoranEngelund committed:
Removed UI component and added UI filter component ->

Removed: Searchbar from header
Added: CTA for sorting by title
Added: CTA for sorting by date

JoranEngelund committed:
added styling for comment, comment-section

Added: border-top for whole collapse-comment section
Added: padding, border-radius for comment

JoranEngelund committed:
Added collapse comment form and static comments

Added: collapse element for comment section
Added: form for commenting on post with cta to submit
Added: Static example of comment with image of user who posted comment

JoranEngelund committed:
added comment section on post-specific

JoranEngelund committed:
Merge pull request #4 from JoranEngelund/new-ui-implementation

New UI implementation

JoranEngelund committed:
created js sub-folder and index.mjs file

JoranEngelund committed:
added script file, id on card for register and method on form

Added: script file
Added: id on card for register
Added: method="POST" in form

JoranEngelund committed:
created contants.mjs that handles urls

created contants.mjs that handles urls for registration and login paths

JoranEngelund committed:
Created registration.mjs ->

Added: function to retrieve values from a form and sends a HTTP POST request to the server. the request contains a object that is created from the form.

Imported: registration api from constants.mjs
Imported: async fetch function that sends registration POST request to server

JoranEngelund committed:
Created error-messages/reg-erros.mjs ->

Added: Function that checks whether a http request returned a false or true response, and then displayes either validation message og error message based on the return value from the request

JoranEngelund committed:
Created handlers/register.mjs ->

Added: Async fetch function that sends a HTTP POST request to the API server.

Imported: registrationError function that checks for errors from response object

JoranEngelund committed:
imported and called function registerUser()

JoranEngelund committed:
changed const form to formData ->

Change: changed variable name that collects data from the form to be named formData

JoranEngelund committed:
updated content on index.html ->

Added: id on card element for loginform
Added: better title explanation of the form validation to email imput
Added: js.file in script tag

JoranEngelund committed:
removed console.logs from function

JoranEngelund committed:
created login.mjs ->

Created: form-eventlistener to login form, for saving values from inputs, and send those to the api server for validation

JoranEngelund committed:
Updated new constants ->

Added: login variable with path for login
Added: login api with correct path
Added: posts variable with path for posts
Added: posts api url for fetching posts

JoranEngelund committed:
Created error-message handling for login

JoranEngelund committed:
Updated error/validation message for registration

JoranEngelund committed:
Created login.mjs to handle fetch calls with http requests and tokens

JoranEngelund committed:
removed console loggers from function

JoranEngelund committed:
added localStorage functions

JoranEngelund committed:
Added path handler in index.mjs ->

Imported: loginUser
Added: If statement to handle path, and run the correct function based on the path

JoranEngelund committed:
Updated styling ->

Changed: collapse-comment from ID to class
Added: media query for post cards

JoranEngelund committed:
Added flags on posts endpoints

JoranEngelund committed:
Created getPosts ->

Created: Async function to retrieve posts from api

JoranEngelund committed:
Created getPosts.mjs ->

Import: API for posts endpoint
Import: getPosts from handlers to fetch posts
Added: Posts url as argument to getPosts fetch function

JoranEngelund committed:
Added template for creating posts

JoranEngelund committed:
Imported displayPosts, to display all posts on feed

JoranEngelund committed:
Linked index.mjs script file

JoranEngelund committed:
Updated code structure

JoranEngelund committed:
forEach method on comments arguments, with html tags

JoranEngelund committed:
Update getPosts.mjs

JoranEngelund committed:
Updated createPost function ->

Added: if else statement to check if avatar returns null or empty string, to replace it with a placeholder img

JoranEngelund committed:
Updated createPost.mjs with new features ->

Added: JSDocs for createPosts function
Added: new parameters for function
Added: profile avatar for comments owner
Added: if statement to add placeholder image if avatar url is null or empty
Added: forEach and dynamic html to display reaction symbol and count
Added: count for number of comments and reactions
Added: collapse element to showcase reaction symbols on posts
Added: id for post to comment form

JoranEngelund committed:
Updated async function ->

Added: JSDocs for function
Added: destructured 2 more properties from posts array of objects
Added: destructured _count property to access properties
Added: commenCount, reactionCount, reactions argument to createPosts function

JoranEngelund committed:
Added JSDocs for function

JoranEngelund committed:
updated home.html ->

Added: ID for section and form to send post.
Removed: Static post

JoranEngelund committed:
Created allPosts-error.mjs

Added: error function to display error message to user, for response if not okay
Added: error function to display error message to user if catch block catches any error from the try block

JoranEngelund committed:
Added response function error if response returns false

JoranEngelund committed:
Updated getPosts.mjs file ->

Imported: error function from allPosts-error.mjs and called them inside the async fetch function
Removed: console.logs

JoranEngelund committed:
Updated home.html ->

Added: paragraph element to display error messages.
Added: pattern for tags input aswell with title explanation for better UX

JoranEngelund committed:
Added enpoint for sending posts to api

JoranEngelund committed:
Created submitPost-error file ->

Created: function that checks if response returns false, in case display message to user
Created: Function that checks if any error are catched in the try block, if so display message to user

JoranEngelund committed:
removed console.log

JoranEngelund committed:
created sendPost.mjs file ->

Created: Async function that sends request to server with object passed in as argument

JoranEngelund committed:
Imported and called sendPost() function

JoranEngelund committed:
Created submitPost.mjs file ->

Created: Form event listener that retrieves values from form, parses it into an object and sends it to api server with form method POST

JoranEngelund committed:
changed post to stringify the object

JoranEngelund committed:
Merge pull request #5 from JoranEngelund/js2-ca_joran-engelund

js2 - added functionalities: user can loggin, user can register, user can create post to feed, user can view feed(comments and reactions on posts, along with count of comments and reactions per post)

JoranEngelund committed:
Added else if statement to index.mjs ->

Added: if else statement to check path and then run function associated with those paths

JoranEngelund committed:
Added 3 more function and JSDocs to storage ->

Added: remove function, to removeItem from localStorage
Added: clear function, to clear localStorage of items
Added: JSDocs for all functions

JoranEngelund committed:
Merge pull request #6 from JoranEngelund/js2-ca_joran-engelund

Js2 ca joran engelund - 

Added: else if statement to index.mjs
Added:  3 more function and JSDocs to storage

JoranEngelund committed:
Updated getPosts.mjs ->

Imported: getForm() function that retrieves all comment-forms, function does aswell submit and sends comments to api server

JoranEngelund committed:
Created sendComment.mjs ->

Created: async fetch call that sends comments to api server,

JoranEngelund committed:
updated to check if repsonse.ok ->

Added: if statement that checks if repsonse.ok === true, reload current page when submitting form

JoranEngelund committed:
created comment.mjs ->

Added: function that takes a retrieved comment-form from a try block and sends it into a forEach with a addEventListener, which sends the form data to api server and posts it as a comment

JoranEngelund committed:
Updated createPost.mjs template ->

Added: converter for date .
Added: id on container for edit delete post anchor elements
Added: method on comment-form
Added: class "comment-form" on comment form
Added: required on texteare for comment-form

JoranEngelund committed:
Added export on const API_BASE_URL

JoranEngelund committed:
Added trim() on title value so not to send space

JoranEngelund committed:
added .trim() to body.value to remove empty spaces

JoranEngelund committed:
Merge pull request #7 from JoranEngelund/js2-ca_joran-engelund

Js2 ca joran engelund

dvergnir committed:
Install bootstrap and sass

dvergnir committed:
removed .active class from Home @ post-specific

dvergnir committed:
import function to load specific post and add path

dvergnir committed:
export base url

dvergnir committed:
call api with querystring url and call html funct.

dvergnir committed:
function to deliver url and display post

dvergnir committed:
added class to remove "view post" on specific post

dvergnir committed:
removeda duplicate call to funct getSpecificPost

dvergnir committed:
moved id to container and deleted an extra div

dvergnir committed:
Merge pull request #8 from JoranEngelund/js-2-ca_benjamin-londal

Js 2 ca benjamin londal

JoranEngelund committed:
removed .trim() from body value on comment-form

JoranEngelund committed:
removed .trim() from body value on submit-post form

JoranEngelund committed:
updated createPost template ->

Added: if else statement to check whether owner of post is the same as the logged in user, if, then parse in the edit-delete links to the generated post

JoranEngelund committed:
Updated createPosts.mjs in template ->

Added: removed id from edit-delete container and added edit-post class instead
Added: post id to updatePostModal for edit form
Added whole modal with form for editing posts. targeted with ID of post

JoranEngelund committed:
Created edit.mjs

function that takes a retrieved edit-post-form from a try block and sends it into a forEach with a addEventListener, which sends the form data to api server and edit the post selected

JoranEngelund committed:
updated JSDocs

JoranEngelund committed:
added function to retrieve editForm and commentForm from specificPosts

JoranEngelund committed:
added getEditForm for all posts

JoranEngelund committed:
created editPost.mjs

added: Async fetch call that sends an edit-post to the API endpoint

JoranEngelund committed:
removed modal ->

removed: modal that contains form for editing post. implemented dynamically instead

JoranEngelund committed:
removed modal ->

removed: modal that contains form for editing post. implemented dynamically instead

JoranEngelund committed:
Merge pull request #9 from JoranEngelund/joran-engelund_js2-ca

Joran engelund js2 ca

JoranEngelund committed:
updated login-error

Added: clearing of storage if response.ok is false and also new error-message that removes the form and presents a cta to make the user try again, sends them to the same page basically. Had an issue where you tried logging in with correct credentials after an error that it took you to an non existing page

dvergnir committed:
installed errormessage for specific post

JoranEngelund committed:
added class on anchor tag for delete link

JoranEngelund committed:
created delete.mjs

function that retrieves an link/cta for a post, that let's you delete the post. -> it will loop through all the delete links, and then adds a eventListener on "click" where it retrieves the ID from the clicked element that should be the same ID as a created post. It then uses the ID from the clicked element and passes it into a API endpoint, which then is used as a argument for a delete fetch call

JoranEngelund committed:
created deletePost.mjs ->

async fetch call that deletes a post based on the Post ID

JoranEngelund committed:
updated getPosts.mjs ->

imported: getDelete function
Added: querySelector to retrieve the delete-post element from all posts
Added: Called the getDelete function and passed in the delete element retrieved

JoranEngelund committed:
Updated getSpecificPost.mjs

Imported: getDelete function
Added: querySelectorAll to retrieve all the .delete-post elements from posts
Added: called the getDelete() and passed in the delete-post element as argument

JoranEngelund committed:
added styling for reaction elements

JoranEngelund committed:
Updated createPosts.mjs

Changed: reactionHTML to separate the reaction symbol and the reaction count for the symbol. making it possible to send the symbol as a reaction
Added: user Avatar to comment form. added placeholder of avatar isn't specified
added collaps and a set of symbols/emojies to react with on posts

JoranEngelund committed:
Update styles.css

JoranEngelund committed:
Updated login-error message ->

Changed: structure on error-message info displayed to user

JoranEngelund committed:
updated getPosts.mjs

Imported: reactionListener function
Added: querySelectorAll on reaction elements in posts
Added: Called function reactionListener and sent reaction elements as argument

JoranEngelund committed:
Created reactionPost.mjs

Created: Async fetch function that sends a URL endpoint request with postId and symbol string to the server

JoranEngelund committed:
created reaction.mjs with reactionListener function ->

Created: function that listens on click and then retrieves the reaction elements on, and will then store the elements .innerText and ID in a URL which will be sent to the API server

JoranEngelund committed:
updated getSpecificPost.mjs ->

Imported: reactionListener function
Added: querySelectorAll to retrieve all reaction elements on posts
Added: Called reactionListener and passed in reaction elements as argument

dvergnir committed:
removed dummypost from home.html

dvergnir committed:
part of restructuring to pure functions and setup

dvergnir committed:
part of restructuring to pure functions

JoranEngelund committed:
added title and body values in edit post form

JoranEngelund committed:
Merge pull request #10 from JoranEngelund/js2-ca_joran

Js2 ca joran

dvergnir committed:
part of restructuring to pure functions

dvergnir committed:
created setup

dvergnir committed:
deleted unused function

dvergnir committed:
added functions  to resolve conflict in PR

dvergnir committed:
moved files

JoranEngelund committed:
Merge branch 'js2' into benjamin-londal_js2-ca

dvergnir committed:
Merge pull request #11 from JoranEngelund/benjamin-londal_js2-ca

Benjamin londal js2 ca

JoranEngelund committed:
removed import sendPost

JoranEngelund committed:
removed spaces and added ;

JoranEngelund committed:
Updated comment.mjs ->

Changed: JSDocs description
Changed: function name from getForm -> commentListener
Added: querySelector to grab comment forms

JoranEngelund committed:
Updated delete.mjs ->

Changed: JSDocs description
Changed: function name from getDelete -> deleteListener
Added: querySelectorAll to grab delete-post element

JoranEngelund committed:
Updated edit.mjs

Changed: JSDocs description
Changed: function name from getEditForm -> editListener
Added: querySelectorAll to grab edit-forms

JoranEngelund committed:
Update reaction.mjs

Changed: JSDocs description
Added: querySelectorAll to grab .send-reaction elements

JoranEngelund committed:
removed imports, function calls and variables ->

everything has been moved to new files for better structure

All right reserved - Benjamin Løndal

JoranEngelund committed:
created listeners.mjs ->

Created listeners function to nest all post functionalities for better structure

All rights reserved - Benjamin Løndal

JoranEngelund committed:
imported listeners function from listeners.mjs and called it in setup function

all rights reserved - Benjamin Løndal

JoranEngelund committed:
changed imports and functions ->

Change: Imports and function names has been changed in regards of the new functions names and structure. Variables with querySelectors has been removed aswell

JoranEngelund committed:
Merge pull request #12 from JoranEngelund/js2-ca_joran

Js2 ca joran

dvergnir committed:
creater searchlistener and filter function

dvergnir committed:
added argument to searchListener() in setup()

JoranEngelund committed:
Update README.md

dvergnir committed:
Adds JSDocs to new functions
removed unused const

dvergnir committed:
Merge pull request #13 from JoranEngelund/js2-ca_benjamin

Js2 ca benjamin

dvergnir committed:
created html page for specific profiles

dvergnir committed:
added classes to target elements for dynamic html

dvergnir committed:
added profile url

dvergnir committed:
reverted changes done trying to reuse api call

dvergnir committed:
same as prior

dvergnir committed:
built profile template function

dvergnir committed:
profile function (not in use yet)

dvergnir committed:
built file to display my profile

JoranEngelund committed:
Updated styling for tags dropdown and post images

JoranEngelund committed:
Updated home.html and created 2 shadow pages for sort/filter of posts

Added: dropdown menus for filter/sort on home.html
Created: shadow page for filtered tags
Created: shadow page for sorted posts

JoranEngelund committed:
Created 3 functions for filtering of tags ->

FilterTags.mjs: created function that retrieves queryString from the url of the current page it's on and passes it into a API endpoint fetch call, to filter and show the tag string name that's in the parameter value

filter-dropdown.mjs: function that retrieves all tags from api call, maps it into a new array and generates it dynamically in a dropdown menu

filteredTagPosts.mjs: Async fetch call the sends a GET request to the server with the provided URL, returning the posts filtered by one tag name

JoranEngelund committed:
Update getSpecificPost.mjs

JoranEngelund committed:
changes reactions to likes to avoid horizontal scrollbar

JoranEngelund committed:
removed console.log from setup.mjs

JoranEngelund committed:
created 2 functions for sorting ->

sortType.mjs: Function that retrieves queryString from the url of the current page it's on and passes it into a API endpoint fetch call, to sort and show the posts in the sorted order that's in the parameter value

sortPost.mjs: Async fetch call that sorts the posts based on the URL endpoint with the neccessary sort="typeOfSort" & sortOrder="sortOrder" flags. returns and displays the posts sorted

JoranEngelund committed:
imported filterTags and sortPosts to index.mjs

JoranEngelund committed:
imported and called filterTemplateTags to retrieve the tags from the api call

dvergnir committed:
reworked file management for profile setup

JoranEngelund committed:
Added sign-out class to sign out link in navigation

JoranEngelund committed:
created logout.mjs ->

Added: Function that listens to user click on "signout" link in navigation, if link is clicked, the localStorage gets cleared of items

JoranEngelund committed:
Imported and called signout() function per page

dvergnir committed:
removed dummy html and prepared for dynamic feed

dvergnir committed:
added author to search function

dvergnir committed:
loading posts to profile feed, need to fix errs

JoranEngelund committed:
Update getPosts.mjs

JoranEngelund committed:
removed console.log

JoranEngelund committed:
re-structured functions and files for getSpecificPost

dvergnir committed:
minor work and debugging on profile

JoranEngelund committed:
restructured som sorting function

JoranEngelund committed:
changes added benjii

JoranEngelund committed:
added defined values from post to edit form, so user can view what they already have in the post, when editing

JoranEngelund committed:
removed unnecessary code

JoranEngelund committed:
removed commented out import and function call

dvergnir committed:
Merge pull request #15 from JoranEngelund/js2-ca_benjamin-londal-TEST

changes added benjii

dvergnir committed:
removed dummy content and minor debug

dvergnir committed:
Merge branch 'js2-ca_benjamin-londal' of https://github.com/JoranEngelund/Social-Media-Application into js2-ca_benjamin-londal

JoranEngelund committed:
added if statement to relocate user to home page if response.ok returns false

JoranEngelund committed:
Merge pull request #14 from JoranEngelund/js2-ca_Engelund-Joran

Js2 ca engelund joran

dvergnir committed:
avoid stacking personal posts on top of home feed

JoranEngelund committed:
removed some html elements which messed with responsivness of profile elements

dvergnir committed:
bootstrap update

dvergnir committed:
profile work

dvergnir committed:
Merge branch 'js2-ca_benjamin-londal' of https://github.com/JoranEngelund/Social-Media-Application into js2-ca_benjamin-londal

dvergnir committed:
added value to form variable

JoranEngelund committed:
Revert "Js2 ca engelund joran"

JoranEngelund committed:
Merge pull request #17 from JoranEngelund/revert-14-js2-ca_Engelund-Joran

Revert "Js2 ca engelund joran"

dvergnir committed:
Merge pull request #16 from JoranEngelund/js2-ca_benjamin-londal

Js2 ca benjamin londal

JoranEngelund committed:
Needed to re-add reverted features because of merge-conflict

JoranEngelund committed:
Updated styling to prevent horizontal scrollbar, chenged post-img to max-width 100%

dvergnir committed:
calling specific profile

JoranEngelund committed:
Added wrap on reaction emojies container->

JoranEngelund committed:
Added dynamic link to profile

JoranEngelund committed:
Update README.md

JoranEngelund committed:
Update README.md

JoranEngelund committed:
Update README.md

JoranEngelund committed:
Update README.md

JoranEngelund committed:
Update README.md

JoranEngelund committed:
Merge pull request #19 from JoranEngelund/js2-CA_Engelund

Js2 ca engelund

dvergnir committed:
set up profile-specific to view profiles by title

JoranEngelund committed:
Update README.md

dvergnir committed:
replaced src variable

dvergnir committed:
restructured specific profile for consistency

dvergnir committed:
removed follow button on user profile

dvergnir committed:
further file management fixes

dvergnir committed:
minor fix

dvergnir committed:
fixed url to use variable

JoranEngelund committed:
Added alert messages to better UX ->

Added: alert messages on Edit, Delete, Edit profile media, reaction and send comment handlers for better UX

JoranEngelund committed:
adjusted a little position on loadingIndicator for profile page

JoranEngelund committed:
added loadingIndicator at profile.html

JoranEngelund committed:
Imported and called toggleLoadingIndicator

JoranEngelund committed:
fixed dynamic change on profile-link in comment section to redirect either to logged in profile or specific-profile

JoranEngelund committed:
Merge pull request #20 from JoranEngelund/js2-ca_Joran-Engelund

Added alert messages to better UX ->

dvergnir committed:
Started follow function

dvergnir committed:
JSDocs

JoranEngelund committed:
implemented placeholder img on broken image src

JoranEngelund committed:
Delete 279036893_10158362550777032_8021443828468629015_n.jpg

JoranEngelund committed:
updated object listeners

JoranEngelund committed:
Needed to re-structure and re-factor filter function as it crashed with searchListener

dvergnir committed:
further work on follow function

dvergnir committed:
further follow work

dvergnir committed:
follow changes

dvergnir committed:
code cleanup and JSDocs

JoranEngelund committed:
Merge branch 'js2' into js2-ca_benjamin-londal

JoranEngelund committed:
Merge pull request #22 from JoranEngelund/js2-ca_benjamin-londal

Js2 ca benjamin londal

JoranEngelund committed:
Merge branch 'js2' into js2-ca_Joran-Engelund

JoranEngelund committed:
Merge pull request #21 from JoranEngelund/js2-ca_Joran-Engelund

Js2 ca joran engelund

JoranEngelund committed:
removed som features that was added again after merge-conflict

JoranEngelund committed:
Updated createProfile.mjs

Removed: double up innerHTML that got carried over from merge-conflict
Added: dynamic profile link for avatars in following/follower section

JoranEngelund committed:
added loadingIndicator on specific-profile.mjs

JoranEngelund committed:
needed to re-structure listeners for posts on profile

JoranEngelund committed:
Clean-up code

JoranEngelund committed:
Update follow.mjs

JoranEngelund committed:
added aspect-ration on profile-images

Fermain reviewed (commented):
The required and additional features are in place and working without error. This application passes the mustard but there is still areas that you could improve. You may not want to do them now in the interest of sticking with the deadline but consider for future projects or a refactor of this project later down the line.

1. File structure

In terms of file structure, you could use folders to structure HTML so that less is contained in the root folder. It is common for HTML files to form a hierarchy:

`/profile/settings/media/`
`/user/Oliver/followers/`

The URLs can only be constructed by arranging HTML files in hierarchy.

2. Division of Responsibility

In code files there is a tendency towards functions that hold too much responsibility. For example, an API function might request data, then render that data, then set listeners for the rendered HTML. Ideally each function does one task and one task only, so that it can be tested and reused more easily.

JoranEngelund committed:
Needed to create own script file to handle and run loginUser function as not to crash when host+port didn't have any pathname

JoranEngelund committed:
fixed issue on routing problem ->

Fixed issue on routing problem where the login form crashed when there was no pathname === "index.html" in the url

JoranEngelund committed:
added placeholder img on profile avatars if url src fails to load

JoranEngelund committed:
Re-factoring and re-structuring filter ->

Re-factoring and re-structuring of the filter functionality.

JoranEngelund committed:
removed unused imports from functions ->

removed unused imports from ->
setup.mjs
getSpecificPost.mjs
render.mjs

JoranEngelund committed:
Imported and called new functions name in routher

JoranEngelund committed:
re-structur, re-factoring of sorting functions

JoranEngelund committed:
Added routing to login page if token is not defined

JoranEngelund committed:
created router.mjs

wrapped routing in function and exported to index.mjs and then called the router there

JoranEngelund committed:
Merge pull request #25 from JoranEngelund/js2-ca-extra-Joran

Js2 ca extra joran

JoranEngelund committed:
Update README.md
